modular_skyrat/modules/basic_mobs/code/chinchilla.dm

```diff
@@ -0,0 +1,100 @@
+/mob/living/basic/pet/chinchilla
+	name = "chinchilla"
+	desc = "They're a chinchilla."
+
+	icon = 'modular_skyrat/master_files/icons/mob/newmobs.dmi'
+	held_lh = 'modular_skyrat/master_files/icons/mob/inhands/pets_held_lh.dmi'
+	held_rh = 'modular_skyrat/master_files/icons/mob/inhands/pets_held_rh.dmi'
+	icon_state = "chinchilla_white"
+	icon_living = "chinchilla_white"
+	icon_dead = "chinchilla_white_dead"
+
+	maxHealth = 10
+	health = 10
+	mob_size = MOB_SIZE_TINY
+
+	speed = 1.25 //speedy little fuckers
+	see_in_dark = 6
+
+	butcher_results = list(/obj/item/food/meat/slab = 1)
+
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "kick"
+	response_harm_simple = "kicks"
+
+	gold_core_spawnable = FRIENDLY_SPAWN
+
+	can_be_held = TRUE
+	held_w_class = WEIGHT_CLASS_TINY
+	held_state = "chinchilla_white"
+
+	///In the case 'melee_damage_upper' is somehow raised above 0
+	attack_verb_continuous = "bites"
+	attack_verb_simple = "bite"
+	attack_sound = 'sound/weapons/bite.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE
+
+	ai_controller = /datum/ai_controller/basic_controller/chinchilla
+
+	var/body_color
+
+/mob/living/basic/pet/chinchilla/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/pet_bonus, "squeaks happily!")
+
+	if(isnull(body_color))
+		body_color = pick("black", "white")
+		held_state = "chinchilla_[body_color]" // not handled by variety element
+		AddElement(/datum/element/animal_variety, "chinchilla", body_color, FALSE)
```

Comment:
This way, any subtype you do will simply use `animal_variety` so you don't have to do what you did for the black chinchilla variant.
```suggestion
	if(isnull(body_color))
		body_color = pick("black", "white")

	held_state = "chinchilla_[body_color]" // not handled by variety element
	AddElement(/datum/element/animal_variety, "chinchilla", body_color, FALSE)
```